### PR TITLE
Check use-octavia property on CCM configMap

### PR DESCRIPTION
**Pull request by rlobillo**
Covers https://bugzilla.redhat.com/show_bug.cgi?id=2074471

### CODE REVIEW

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [inline test/extended/openstack/cloud-provider-config.go:149]: We could probably put this into the function itself, so that we don't repeat it in line 152?

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [inline test/extended/openstack/cloud-provider-config.go:125]: I know it's a pattern in this file, nevertheless we're starting to mix snake_case with camelCase in the project, this is awful, let's just use camelCase everywhere.

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [inline test/extended/openstack/cloud-provider-config.go:169]: Why not make it simple?

```
			e2e.Logf("Checking config for networkType detected: %q", network.Status.NetworkType)
			expected := userPropertyValue
			if network.Status.NetworkType == "Kuryr" {
				expected = "true"
			}

			o.Expect(ccmPropertyValue).Should(o.Equal(expected),
				"Unexpected value for property %q on section %q. configMap '%q', namespace '%q' with NetworkType=%s.",
				propertyName, sectionName, ccmConfigMap.name, ccmConfigMap.namespace, network.Status.NetworkType)
			e2e.Logf("  - Property %q with correct value %q on section %q in configMap %q (namespace: %q) with NetworkType=%s.",
				propertyName, ccmPropertyValue, sectionName, ccmConfigMap.name, ccmConfigMap.namespace)
```

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [inline test/extended/openstack/cloud-provider-config.go:193]: I think I'd add error handling here directly so that we don't need to repeat it when running this function. It's private anyway.

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [inline test/extended/openstack/cloud-provider-config.go:158]: Wait, I have an impression it should be "false" when Kuryr is running. Yet this passes tests, so I'm confused now.

**Review by dulek — CHANGES_REQUESTED** [review]
Looks good, some style remarks and one doubt inline.

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [reply test/extended/openstack/cloud-provider-config.go:125]: Ah, camelCase is also preferred by golang in general: https://go.dev/doc/effective_go#mixed-caps

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [reply test/extended/openstack/cloud-provider-config.go:158]: Yeah, seems like test does nothing.

File: test/extended/openstack/cloud-provider-config.go
Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: I prefer not to through errors on functions and handle them on the test itself. I believe this is better for debugging failing testcases.

File: test/extended/openstack/cloud-provider-config.go
Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: I changed it all along the file. Thanks! Will do like that from now on

File: test/extended/openstack/cloud-provider-config.go
Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: Thanks for the suggestion. I changed the logic a little bit. I included also the "enabled" property in the checks.

File: test/extended/openstack/cloud-provider-config.go
Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: It is private but can be reused for other future tests in the file. As I mention before, I prefer to see the exceptions on the test itself, it is better for debugging.

File: test/extended/openstack/cloud-provider-config.go
Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: According to the logs I am writting and the different tests I did, it is working fine:

```
$ make openstack-tests && KUBECONFIG=~/ostest/auth/kubeconfig ./openstack-tests run-test "[sig-installer][Suite:openshift/openstack] The Openshift on cloud provider configuration should update use-octavia and enabled properties depending on the NetworkType [Suite:openshift/conformance/parallel]"
go build -o openstack-tests ./cmd/openshift-tests
[BeforeEach] [Top Level]
  /home/stack/openstack-test/vendor/github.com/openshift/origin/test/extended/util/test.go:61
[BeforeEach] on cloud provider configuration
  /home/stack/openstack-test/test/extended/openstack/cloud-provider-config.go:31
STEP: preparing openshift dynamic client
[It] should update use-octavia and enabled properties depending on the NetworkType [Suite:openshift/conformance/parallel]
  /home/stack/openstack-test/test/extended/openstack/cloud-provider-config.go:109
Jul 26 16:15:54.032: INFO: Checking config for networkType detected: "Kuryr"
Jul 26 16:15:54.032: INFO: Properties with correct values on section "LoadBalancer" in configMap "cloud-conf" (namespace: "openshift-cloud-controller-manager") with NetworkType=Kuryr.
Jul 26 16:15:54.032: INFO: - Properties (["use-octavia" "enabled"]): Values (["true" "false"])

```

Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: the ccm config map is setting use-octavia=True and enabled=False. This is what I understood from the BZ and the PR fixing it (https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/183)

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [reply test/extended/openstack/cloud-provider-config.go]: Okay.

File: test/extended/openstack/cloud-provider-config.go
Comment (dulek) [inline test/extended/openstack/cloud-provider-config.go:160]: Okay, so IIUC this only makes sense once we make a switch to use CCM. If so I would wait with merging this test until that happens as we're cautious with the possibility of it slipping again.

Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go:160]: In my local tests, I am not testing with external CCM enabled in the cluster, however the CCM configMap is configured. If you want to use this test to confirm that the fix implemented for https://bugzilla.redhat.com/show_bug.cgi?id=2074471 remains in place, I would merge it. But, if you think that behaviour could change, I am ok waiting until 4.12 enables CCM by default.

Comment (dulek) [reply test/extended/openstack/cloud-provider-config.go:160]: Okay, I see the point, it's more about CCM bug even though CCM is just tech preview. Cool, I'm fine with this then, but I'll let @MaysaMacedo take a look too.

**Comment by dulek** [comment]
/lgtm

Leaving /approve for a second pair of eyes.

File: test/extended/openstack/cloud-provider-config.go
Comment (MaysaMacedo) [inline test/extended/openstack/cloud-provider-config.go:163]: When Kuryr is not used, we fix the value of use-octavia as well, to always be true since lbaasv2 is not supported. Perhaps this test can be updated to take that into account?
https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/pkg/cloud/openstack/openstack.go#L85-L93

Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go:163]: Hello @MaysaMacedo, in such a case the property use-octavia is useless on the cloud-provider-config?

What should happen if use-octavia = false or the property is not present in cloud-provider-config?

Comment (MaysaMacedo) [reply test/extended/openstack/cloud-provider-config.go:163]: cluster-cloud-controller-manager (CCCMO) enforces that configuration.
I'm proposing the removal of it from the installer as is not needed https://github.com/openshift/installer/pull/5834

Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go:163]: Great! Thanks, I will update the test accordingly. No need to check the cloud-provider-config then.

on the eCCM cm, for Kuryr: 

`expectedValues = []string{"true", "false"}`

for any other networkType:

`expectedValues = []string{"true", "#UNDEFINED"}`

Comment (MaysaMacedo) [reply test/extended/openstack/cloud-provider-config.go:163]: I think we still need it. If the user changes the user managed configmap, the internal one will continue to enforce the appropriate values.

Comment (rlobillo) [reply test/extended/openstack/cloud-provider-config.go]: Thanks @MaysaMacedo I updated the name of the test and updated the expected values for NetworkType != Kuryr to fixed values. 

I also tested locally with OpenShiftSDN and Kuryr and in both cases the testcase is passing.

Please take a look and let me know if you think something is missing.

**Comment by MaysaMacedo** [comment]
/lgtm
While I saw the test passed, I didn't see any logs of the steps within the test. Is this expected?

**Comment by rlobillo** [comment]
> /lgtm While I saw the test passed, I didn't see any logs of the steps within the test. Is this expected?

Hello @MaysaMacedo 

Checking [0], the steps are not written for any test. I checked for already running tests and it is not there either. If you want to see the steps you need to run the test isolated, like:

`openshift-test run-test "<name_of_the_test>"`

I think this is the expected behaviour.

[0] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_openstack-test/44/pull-ci-openshift-openstack-test-main-test-kuryr/1556886463394615296/artifacts/test-kuryr/openstack-test-openstack/artifacts/e2e.log

**Comment by MaysaMacedo** [comment]
okay, thanks!
/approve

**Comment by MaysaMacedo** [comment]
/hold
There is something wrong with a kuryr test case, which is unrelated to this test scenario. I'm taking a look

**Pull request merged**
